server/repository/src/database_settings.rs

```diff
@@ -56,10 +64,64 @@ impl DatabaseSettings {
     }
 }
 
+// feature sqlite
+#[cfg(not(feature = "postgres"))]
+#[derive(Debug)]
+pub struct SqliteConnectionOptions {
+    pub enable_wal: bool,
+    pub enable_foreign_keys: bool,
+    pub busy_timeout_ms: Option<u32>,
+}
+// feature sqlite
+#[cfg(not(feature = "postgres"))]
+impl diesel::r2d2::CustomizeConnection<SqliteConnection, diesel::r2d2::Error>
+    for SqliteConnectionOptions
+{
+    fn on_acquire(&self, conn: &mut SqliteConnection) -> Result<(), diesel::r2d2::Error> {
+        (|| {
+            if self.enable_wal {
+                // println!("Adding WAL");
+                let _r =
+                    conn.batch_execute("PRAGMA journal_mode = WAL; PRAGMA synchronous = NORMAL;");
+                // println!("{:?}", r);
+            }
+            if self.enable_foreign_keys {
```

Comment:
what would be use case for not enabling them (are they not enabled by default ?)

Reply:
It turns out they are not!

This code was stolen from a stackoverflow post, so I have currently copied their configurable approach.
For our use case it might make sense to just hard code here?

https://stackoverflow.com/questions/57123453/how-to-use-diesel-with-sqlite-connections-and-avoid-database-is-locked-type-of

I assume we always want things like WAL too, unless it needs different settings for android or something.
Being able to tweak things like locktimeout without a rebuild might be useful in some production setting one day?

------
"Foreign key constraints are disabled by default (for backwards compatibility), so must be enabled separately for each [database connection](https://www.sqlite.org/c3ref/sqlite3.html). (Note, however, that future releases of SQLite might change so that foreign key constraints enabled by default. Careful developers will not make any assumptions about whether or not foreign keys are enabled by default but will instead enable or disable them as necessary.)"

- https://www.sqlite.org/foreignkeys.html

Reply:
yes I would say hard code config would be enough for our use